network/dnos10/dnos10_template.py

```diff
@@ -0,0 +1,176 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+DOCUMENTATION = """
+---
+module: dnos10_template
+version_added: "2.2"
+author: "Senthil Kumar Ganesan (@skg_net)"
+short_description: Manage Dell OS10 device configurations over SSH.
+description:
+  - Manages Dell OS10 network device configurations over SSH.  This module
+    allows implementors to work with the device running-config.  It
+    provides a way to push a set of commands onto a network device
+    by evaluating the current running-config and only pushing configuration
+    commands that are not already configured.  The config source can
+    be a set of commands or a template.
+extends_documentation_fragment: dnos10
+options:
+  src:
+    description:
+      - The path to the config source.  The source can be either a
+        file with config or a template that will be merged during
+        runtime.  By default the task will first search for the source
+        file in role or playbook root folder in templates unless a full
+        path to the file is given.
+    required: true
+  force:
+    description:
+      - The force argument instructs the module not to consider the
+        current device running-config.  When set to true, this will
+        cause the module to push the contents of I(src) into the device
+        without first checking if already configured. This argument is
+        mutually exclusive with O(config).
+    required: false
+    default: false
+    choices: [ "true", "false" ]
+  backup:
+    description:
+      - When this argument is configured true, the module will backup
+        the running-config from the node prior to making any changes.
+        The backup file will be written to backup_{{ hostname }} in
+        the root of the playbook directory. This argument is
+        mutually exclusive with O(config).
+
+    required: false
+    default: false
+    choices: [ "true", "false" ]
+  config:
+    description:
+      - The module, by default, will connect to the remote device and
+        retrieve the current running-config to use as a base for comparing
+        against the contents of source.  There are times when it is not
+        desirable to have the task get the current running-config for
+        every task.  The I(config) argument allows the implementer to
+        pass in the configuration to use as the base config for
+        comparison. This argument is mutually exclusive with
+        O(force) and O(backup).
```

Comment:
I meant both, since if the user specifies config and we are using that to compare and applying the config. It doesn't make sense to backup the current running config.